### ISSUE TITLE
[REPL] broken tab completion layout for methods

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#45836]
```
julia> include(<tab>
include(mapexpr::Function, fname::AbstractString)
     @ Base.MainInclude client.jl:469                                                    include(fname::AbstractString)
     @ Base.MainInclude client.jl:470
```

expected (from v1.8):
```
julia> include(<tab>
include(mapexpr::Function, fname::AbstractString) in Base.MainInclude at client.jl:473  include(fname::AbstractString) in Base.MainInclude at client.jl:474
```

**Comment by Seelengrab** [comment]
I was not aware of tab completion falling back to method printing directly - I'd have assumed it had more custom printing, since this should have been a problem with longer function signatures already, right?

**Comment by Liozou** [comment]
That's frankly understandable since there was no test on the printing of  multicolumn REPL completion. The fix for this issue at #45844 (awaiting review) includes one.

It was not actually a problem with long function signatures because multi-column printing is decided depending on the width of the largest completion, so if one was too large it reverts to being on a single column. The only implicit assumption that #45069 broke was that no completion would include a newline, but that should have been tested.
